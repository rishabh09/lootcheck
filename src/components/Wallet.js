import React from 'react'
import {connect} from 'react-redux'
import {depositMoney, withdrawMoney} from '../actions/balance'
import {WITHDRAW} from '../actions/constants'

export class Wallet extends React.Component {
  constructor () {
    super()
    this.state = {balance: undefined}
  }

  updateBalance = e => this.setState({balance: parseInt(e.target.value, 10)})

  deposit = () => this.props.depositMoney(this.state.balance)
  withdraw = () => this.props.withdrawMoney(this.state.balance)
  render () {
    return (
      <div>
        <h3 className='balance'>Your Balance is {this.props.balance}</h3>
        <br />
        <input className='input-balance' onChange={this.updateBalance} />
        <button className='btn-deposit' onClick={this.deposit}>DEPOSIT</button>
        <button className='btn-withdraw' onClick={this.withdraw}>WITHDRAW</button>

      </div>
    )
  }
}

export default connect(state => { return {balance: state} }, {depositMoney, withdrawMoney})(Wallet)
