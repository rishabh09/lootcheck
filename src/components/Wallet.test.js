import React from 'react'
import {shallow} from 'enzyme'
import {Wallet} from './Wallet'

describe('wallet component', () => {
  const props = {balance: 20}
  const wallet = shallow(<Wallet {...props} />)
  it('should render wallet', () => {
    expect(wallet).toMatchSnapshot()
  })
  it('should display balance properly', () => {
    expect(wallet.find('.balance').text()).toEqual('Your Balance is 20')
  })
})
