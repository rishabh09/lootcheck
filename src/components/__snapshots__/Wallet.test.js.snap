// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wallet component should render wallet 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={20}
    depositMoney={[Function]}
    withdrawMoney={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="balance"
        >
          Your Balance is 
          20
        </h3>,
        <br />,
        <input
          className="input-balance"
          onChange={[Function]}
        />,
        <button
          className="btn-deposit"
          onClick={[Function]}
        >
          DEPOSIT
        </button>,
        <button
          className="btn-withdraw"
          onClick={[Function]}
        >
          WITHDRAW
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Your Balance is ",
            20,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Your Balance is ",
          20,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-balance",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "DEPOSIT",
          "className": "btn-deposit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "DEPOSIT",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "WITHDRAW",
          "className": "btn-withdraw",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "WITHDRAW",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="balance"
          >
            Your Balance is 
            20
          </h3>,
          <br />,
          <input
            className="input-balance"
            onChange={[Function]}
          />,
          <button
            className="btn-deposit"
            onClick={[Function]}
          >
            DEPOSIT
          </button>,
          <button
            className="btn-withdraw"
            onClick={[Function]}
          >
            WITHDRAW
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Your Balance is ",
              20,
            ],
            "className": "balance",
          },
          "ref": null,
          "rendered": Array [
            "Your Balance is ",
            20,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-balance",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "DEPOSIT",
            "className": "btn-deposit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "DEPOSIT",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "WITHDRAW",
            "className": "btn-withdraw",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "WITHDRAW",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
